packages/passenger/rpm/passenger-analytics-collection-static-sleep.patch

```diff
@@ -0,0 +1,66 @@
+--- a/src/agent/Core/ApplicationPool/Pool/AnalyticsCollection.cpp    2025-03-28 15:04:27.371198726 +0200
++++ b/src/agent/Core/ApplicationPool/Pool/AnalyticsCollection.cpp    2025-03-28 15:06:10.586606453 +0200
+@@ -23,6 +23,9 @@
+  *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  *  THE SOFTWARE.
+  */
++#include <cstdlib> // getenv
++#include <climits> // INT_MAX
++
+ #include <Core/ApplicationPool/Pool.h>
+ 
+ /*************************************************************************
+@@ -62,8 +65,51 @@
+		}
+ 
+		UPDATE_TRACE_POINT();
+-		unsigned long long currentTime = SystemTime::getUsec();
+-		unsigned long long sleepTime = timeToNextMultipleULL(5000000, currentTime);
++
++		// Open OnDemand override: Change the default sleeping behavior unless the following environment
++		// variable is defined: OOD_OVERRIDE_PASSENGER_ANALYTICS_COLLECTION_RESTORE_UPSTREAM_BEHAVIOR.
++		// Use a static sleep of 5 seconds by default.
++		// This can be overridden with the OOD_OVERRIDE_PASSENGER_ANALYTICS_COLLECTION_SLEEP_TIME_SECONDS variable.
++		// NOTE: Setting this to a very big value will impact the usefulness of passenger-status metrics,
++		//       which will not get gathered unless this loop runs. If you do not care about the metrics,
++		//       feel free to set very large values.
++		const unsigned long long defaultSleepTime = 5000000;
++		unsigned long long sleepTime = defaultSleepTime;
++		const char* envPassengerRestoreUpstreamBehavior = std::getenv("OOD_OVERRIDE_PASSENGER_ANALYTICS_COLLECTION_RESTORE_UPSTREAM_BEHAVIOR");
```

Comment:
Do we realistically see anyone restoring previous behavior? Setting `OOD_OVERRIDE_PASSENGER_ANALYTICS_COLLECTION_SLEEP_TIME_SECONDS` to 5 seconds is more or less the same thing, just without the synchronization, which I have to question if anyone out there actually cares/needs the synchronization.

I guess I'm just wondering if this is a YAGNI situation (you aren't going to need it).

Reply:
I don't really see anyone needing this, unless one wishes to benchmark it perhaps. On the other hand, I'm all for more configurable software, especially when this goes into a compiled program which is harder to modify on the fly (unlike ruby files).

This will be disabled by default, and the only impact of leaving it configurable (as it is now) is that we'll check in each iteration whether the environment variable is defined or not.